mutt/signal.c

```diff
@@ -63,6 +64,41 @@ static sig_handler_t SegvHandler = mutt_sig_exit_handler;
 volatile sig_atomic_t SigInt;   ///< true after SIGINT is received
 volatile sig_atomic_t SigWinch; ///< true after SIGWINCH is received
 
+static void exit_print_int_recursive(int n)
+{
+  char digit;
+
+  if (n > 9)
+    exit_print_int_recursive(n / 10);
+
+  digit = '0' + (n % 10);
+  write(1, &digit, 1);
+}
+
+static void exit_print_int(int n)
+{
+  if (n < 0)
+  {
+    write(1, "-", 1);
+    n = -n;
+  }
+  exit_print_int_recursive(n);
+}
+
+static void exit_print_string(const char *str)
+{
+  size_t len = 0;
+
+  if (!str)
+    return;
+
+  while (str[len])
+    len++;
+
+  if (len > 0)
+    write(1, str, len);
+}
```

Comment:
This all seems like overkill to me :-)
If you'd like to keep it, please can you add Doxygen comments.
Thanks.

Reply:
If I would have written that, I'd have simply called strlen(3).  I just cherry-picked the upstream commit as is.

By overkill, do you mean the entire commit?  Or the way it's implemented?

Reply:
BTW, it would be interesting if gcvt(3) was wide spread; we could just call it.  :(

Reply:
> overkill

A lot of extra code for a situation that "shouldn't happen" :-)

It's the correct thing to do; we should do it.
It's just a shame C can't do it more concisely.

Reply:
Yup.  If gcvt(3) was present in the BSDs, we could use that...  :/